src/generator.cpp

```diff
@@ -752,6 +773,48 @@ GeneratorInternalVariablePtr Generator::GeneratorImpl::generatorVariable(const V
     return internalVariable;
 }
 
+bool Generator::GeneratorImpl::sameOrEquivalentVariable(const VariablePtr &variable1,
+                                                        const VariablePtr &variable2)
+{
+    // Return whether the given variables are the same or are equivalent (be it
+    // directly or indirectly).
+
+    return (variable1 == variable2) || variable1->hasEquivalentVariable(variable2, true);
+}
+
+GeneratorVariablePtr Generator::GeneratorImpl::variableFirstOccurrence(const VariablePtr &variable,
+                                                                       const ComponentPtr &component)
+{
+    // Recursively look for the first occurrence of the given variable in the
+    // given component.
+
+    GeneratorVariablePtr voi = nullptr;
+
+    for (size_t j = 0; j < component->variableCount(); ++j) {
+        VariablePtr testVariable = component->variable(j);
+
+        if (sameOrEquivalentVariable(variable, testVariable)) {
+            voi = GeneratorVariable::create();
+
+            voi->mPimpl->populate(testVariable, component, GeneratorVariable::Type::VARIABLE_OF_INTEGRATION);
+
+            break;
+        }
+    }
+
+    if (voi == nullptr) {
+        for (size_t i = 0; i < component->componentCount(); ++i) {
```

Comment:
same here:
`i < component->variableCount() && voi == nullptr`

Reply:
Good point! Done in 4754c2b.

Reply:
Change is not showing in this diff.

Reply:
Problem of refresh, I guess... :)